### PR TITLE
Fix multiline import specifier sorting

### LINKED ISSUE #51615: Organize imports doesn't keep imports in a vertical column
After reformatting this repo's imports as vertical sorted columns for better merge conflict handling, we noticed that running organize imports produced an odd result. Given this test:

```ts
/// <reference path="fourslash.ts" />

////import {
////    Type1,
////    Type2,
////    func4,
////    Type3,
////    Type4,
////    Type5,
////    Type7,
////    Type8,
////    Type9,
////    func1,
////    func2,
////    Type6,
////    func3,
////    func5,
////    func6,
////    func7,
////    func8,
////    func9,
////} from "foo";
////interface Use extends Type1, Type2, Type3, Type4, Type5, Type6, Type7, Type8, Type9 {}
////console.log(func1, func2, func3, func4, func5, func6, func7, func8, func9);

verify.organizeImports(
`import {
    func1,
    func2,
    func3,
    func4,
    func5,
    func6,
    func7,
    func8,
    func9,
    Type1,
    Type2,
    Type3,
    Type4,
    Type5,
    Type6,
    Type7,
    Type8,
    Type9,
} from "foo";
interface Use extends Type1, Type2, Type3, Type4, Type5, Type6, Type7, Type8, Type9 {}
console.log(func1, func2, func3, func4, func5, func6, func7, func8, func9);`
);
```

Organize imports gives:

```ts
import {
    func1,
    func2, func3, func4, func5,
    func6,
    func7,
    func8,
    func9, Type1,
    Type2, Type3,
    Type4,
    Type5, Type6, Type7,
    Type8,
    Type9
} from "foo";
interface Use extends Type1, Type2, Type3, Type4, Type5, Type6, Type7, Type8, Type9 {}
console.log(func1, func2, func3, func4, func5, func6, func7, func8, func9)
```

### PR DESCRIPTION
**Pull request by andrewbranch**
Fixes #51615

Whenever a codefix/refactor prints a modified node list, the emitter looks at the original nodes to determine whether to preserve newlines between the list elements. We preserve the spacing between two list elements if they were consecutive in the original list and are still consecutive in the modified list. Otherwise, we fall back to default list formatting rules for the parent node type. This logic is pretty good, but the default format for named imports/exports was single-line, so if you sort a shuffled set of named imports that were formatted one per line, you’re going to end up with a sorted list that’s _mostly_ on a single line, but with line breaks preserved between any elements that happened to already be already sorted relative to each other. This is obviously the worst of both worlds.

The fix here is simply to say that if the original list is not single-line, the default list format should be multiline. That way, we can still preserve neighbors that are already well-sorted and grouped on a single line (as we see a couple examples of in `organizeImports1.ts`), and we can preserve any double line breaks that occur between already well-sorted imports, but any shuffles will default to getting their own line.

### CODE REVIEW

**Review by sheetalkamat — APPROVED** [review]

**Comment by andrewbranch** [comment]
Would like to get @rbuckton’s reviews for the minor changes to emit nodes and factory

**Comment by andrewbranch** [comment]
@rubiesonthesky how are you determining that? Can you give an example?

**Review by rbuckton — CHANGES_REQUESTED** [review]

File: src/compiler/emitter.ts [hunk 1]
```diff
@@ -4862,7 +4863,14 @@ export function createPrinter(printerOptions: PrinterOptions = {}, handlers: Pri
     }
 
     function emitList(parentNode: Node | undefined, children: NodeArray<Node> | undefined, format: ListFormat, parenthesizerRule?: ParenthesizerRuleOrSelector<Node>, start?: number, count?: number) {
-        emitNodeList(emit, parentNode, children, format, parenthesizerRule, start, count);
+        emitNodeList(
+            emit,
+            parentNode,
+            children,
+            format | (parentNode ? getAdditionalListFormatFlags(parentNode) : 0),
```
[the remark this answers is not in the document]
Comment (andrewbranch) [reply src/compiler/emitter.ts:4870]: The space is the reason I didn’t use an emit flag. Happy to move it.

File: src/compiler/factory/nodeFactory.ts [hunk 2]
```diff
@@ -4870,7 +4870,7 @@ export function createNodeFactory(flags: NodeFactoryFlags, baseFactory: BaseNode
     // @api
     function updateNamedExports(node: NamedExports, elements: readonly ExportSpecifier[]) {
         return node.elements !== elements
-            ? update(createNamedExports(elements), node)
+            ? update(createNamedExports(elements, node.elements.hasTrailingComma), node)
```
[the remark this answers is not in the document]
Comment (andrewbranch) [reply src/compiler/factory/nodeFactory.ts:4873]: I don’t understand; that was already the case. `createNodeArray` doesn’t copy `hasTrailingComma` from the original elements. The `hasTrailingComma` argument is the canonical value for new node array’s `hasTrailingComma` property. In other words, the node array returned from `createNodeArray` will never have `hasTrailingComma` set to true if you only pass it one argument, as all these functions do.

File: tests/baselines/reference/es6ImportDefaultBindingFollowedWithNamedImportDts.js [hunk 3]
```diff
@@ -101,7 +101,7 @@ export declare var x2: b;
 import { x, a12 as y } from "./server";
 export declare var x4: x;
 export declare var x5: y;
-import { x11 as z } from "./server";
+import { x11 as z, } from "./server";
```
[the remark this answers is not in the document]
Comment (andrewbranch) [reply tests/baselines/reference/es6ImportDefaultBindingFollowedWithNamedImportDts.js:104]: It’s the output. The input had a trailing comma.

**Review by rbuckton — APPROVED** [review]

**Pull request merged**